hasher-matcher-actioner/src/OpenMediaMatch/blueprints/hashing.py

```diff
@@ -38,33 +85,64 @@ def hash_media() -> dict[str, str]:
     """
     media_url = request.args.get("url", None)
     if media_url is None:
-        abort(400, "url is required")
+        abort(400, "Missing required parameter: url")
+
+    if not is_valid_url(media_url):
+        abort(400, "Invalid or unsafe URL provided")
 
-    download_resp = requests.get(media_url, allow_redirects=True, timeout=30 * 1000)
-    download_resp.raise_for_status()
+    # First make a HEAD request to check content length
```

Comment:
The way this is written would allow a malicious server to return a passing response from the HEAD request and a malicious response for the GET; see the requests body limiting docs

Reply:
used `max_content_length` instead.